UI/CMakeLists.txt

```diff
@@ -1,426 +1,373 @@
-if(DISABLE_UI)
-	message(STATUS "UI disabled")
-	return()
-elseif(ENABLE_UI)
-	set(FIND_MODE REQUIRED)
+add_subdirectory(obs-frontend-api)
+
+project(obs)
+
+if(NOT DEFINED TWITCH_CLIENTID
+   OR "${TWITCH_CLIENTID}" STREQUAL ""
+   OR NOT DEFINED TWITCH_HASH
+   OR "${TWITCH_HASH}" STREQUAL ""
+   OR NOT TARGET OBS::browser-panels)
+  set(TWITCH_ENABLED FALSE)
+  set(TWITCH_CLIENTID "")
+  set(TWITCH_HASH "0")
 else()
-	set(FIND_MODE QUIET)
+  set(TWITCH_ENABLED TRUE)
 endif()
 
-if(BROWSER_AVAILABLE_INTERNAL)
-	add_definitions(-DBROWSER_AVAILABLE)
+if(NOT DEFINED RESTREAM_CLIENTID
+   OR "${RESTREAM_CLIENTID}" STREQUAL ""
+   OR NOT DEFINED RESTREAM_HASH
+   OR "${RESTREAM_HASH}" STREQUAL ""
+   OR NOT TARGET OBS::browser-panels)
+  set(RESTREAM_ENABLED FALSE)
+  set(RESTREAM_CLIENTID "")
+  set(RESTREAM_HASH "0")
+else()
+  set(RESTREAM_ENABLED TRUE)
 endif()
 
-add_subdirectory(obs-frontend-api)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/ui-config.h.in
+               ${CMAKE_CURRENT_BINARY_DIR}/ui-config.h)
 
-# ----------------------------------------------------------------------------
+find_package(FFmpeg REQUIRED COMPONENTS avcodec avutil avformat)
+find_package(CURL REQUIRED)
 
-project(obs)
+add_subdirectory(frontend-plugins)
+add_executable(obs)
 
-set(DISABLE_UPDATE_MODULE TRUE CACHE BOOL "Disables building the update module")
+if(BUILD_QT EQUAL 5)
+  find_package(Qt5Network REQUIRED)
+  find_package(Qt5Widgets REQUIRED)
+  find_package(Qt5Svg REQUIRED)
+  find_package(Qt5Xml REQUIRED)
```

Comment:
Alternatively, we could do this:
```
find_package(Qt5 COMPONENTS Network Widgets Svg Xml REQUIRED)
```
https://blog.kitware.com/cmake-finding-qt5-the-right-way/
https://doc.qt.io/qt-5.12/cmake-manual.html

Though I'm not sure if Qt 5.9 or CMake on Ubuntu 18.04 supports this syntax.  We could just punt that change to a later date.
https://doc.qt.io/archives/qt-5.9/cmake-manual.html

Reply:
I'll mark this thread resolved since this was changed.  We could also do this for any other find_package calls for Qt throughout, even if only for a single component.  I just didn't go through and mark them because it's probably less time to search through the text locally.  As I understand it, that's the syntax that Qt recommends nowadays.  So long as it works for Ubuntu 18.04 while we still support it, I'm fine with changing it everywhere.